index.js

```diff
@@ -1,9 +1,33 @@
 'use strict';
+const URL = require('url');
 const urlRegex = require('url-regex');
 const normalizeUrl = require('normalize-url');
+const unescape = require('unescape');
 
 module.exports = (str, opts) => {
-	const urls = str.match(urlRegex());
-	const ret = urls ? urls.map(url => normalizeUrl(url.trim().replace(/\.+$/, ''), opts)) : [];
-	return new Set(ret);
+	opts = opts || {};
+	const ret = new Set();
+	const add = url => ret.add(normalizeUrl(unescape(url.trim().replace(/\.+$/, '')), opts));
+	const urls = str.match(urlRegex()) || [];
+	for (const url of urls) {
+		add(url);
+		if (opts.getQueryParams) {
+			const qsUrls = extractQueryParams(url);
+			for (const qsUrl of qsUrls) {
+				add(qsUrl);
+			}
+		}
+	}
+	return ret;
 };
+
+function extractQueryParams(url) {
+	const ret = new Set();
+	const qs = URL.parse(url, true).query;
+	for (const key of Object.keys(qs)) {
```

Comment:
You need to run the URL through `decodeURIComponent` here in case it's encoded.

Reply:
If you will look at the tests I added you will see it works as I believe that the normalizeUrl library does that for you and the url regex also knows to handle it

Reply:
It's `URL.parse(url, true)` that decodes them, actually. All good then.